### ISSUE TITLE
QgsSnappingConfig.IndividualLayerSettings.setTypeFlag API break in QGIS 3.28+

### ISSUE THREAD

**Issue by ldebek** [qgis/QGIS#52373]
### What is the bug or the crash?

Code (part of the Python plugin code) which was working in QGIS version earlier than 3.28:
` IndividualLayerSettings.setTypeFlag(QgsSnappingConfig.VertexFlag | QgsSnappingConfig.SegmentFlag)`

 is failing now due to following error:
 `TypeError: IndividualLayerSettings.setTypeFlag(): argument 1 has unexpected type 'int' `.
 Workaround is to pass `Qgis.SnappingTypes(Qgis.SnappingType.Vertex | Qgis.SnappingType.Segment)` as a type flag, but that requires extra code for QGIS 3.28+.

### Steps to reproduce the issue

Open QGIS project, select any vector layer in the legend tree, open Python Console and type:
```
layer = iface.activeLayer()
project = QgsProject.instance()
snap_config = project.snappingConfig()
individual_settings = snap_config.individualLayerSettings()
layer_settings = individual_settings[layer]
layer_settings.setTypeFlag(QgsSnappingConfig.VertexFlag | QgsSnappingConfig.SegmentFlag)
```

### Versions

3.28+

### Supported QGIS version

- [x] I'm running a supported QGIS version according to the roadmap.

### New profile

- [x] I tried with a new QGIS profile

**Comment by nyalldawson** [comment]
I've tried to solve this one in the past, and can't find any solution. It's likely an upstream sip bug / missing functionality

**Comment by nyalldawson** [comment]
This is fixed on 3.36+
